docker/errors.go

```diff
@@ -44,3 +44,16 @@ func httpResponseToError(res *http.Response, context string) error {
 		return perrors.Errorf("%sinvalid status code from registry %d (%s)", context, res.StatusCode, http.StatusText(res.StatusCode))
 	}
 }
+
+// generateHttpResponseError check if the error is UnexpectedHTTPResponseError and generate the error message
```

Comment:
```suggestion
// generateHttpResponseError creates a Go error from an HTTP error response of a docker/distribution
// registry
```
or something like that?

Also perhaps name this `registryHTTPResponseToError` or something similar, to be consistent but distinct form `httpResponseToError` above.